### PR TITLE
Feature/v1.0.0 beta.1

### PR DESCRIPTION
**Pull request by aleal**
Please check the code and versioning above... I got this solution for get the sqs queue url, but the point is: the message is build outside of the library, so i decided to expose this method, in order to allowing set the QueueUrl on the message created instead of letting the method Publish set it. The problem with the later approach is that you should inform the queue name  every time you call the method and i think it would be better set everything required on message, before calling Publish method. For the enduser this will be transparent. Please let me know if something is bad smelling

### CODE REVIEW

File: aws/aws-sdk-go.v2/client/sqs/client.go [hunk 1]
```diff
@@ -6,21 +6,29 @@ import (
...
 type client struct {
-	client *sqs.Client
+	client    sqsClient
+	queueUrls map[string]*string
 }
 
 // NewClient returns a initialized client
 func NewClient(c *sqs.Client) Client {
-	return &client{c}
+	return &client{c, map[string]*string{}}
```
Comment (cristiandanner) [inline aws/aws-sdk-go.v2/client/sqs/client.go:31]: this map is not being used, why was it included?

File: aws/aws-sdk-go.v2/client/sqs/client.go [hunk 2]
```diff
@@ -43,3 +51,25 @@ func (c *client) Publish(ctx context.Context, input *sqs.SendMessageInput) error
...
+	result, err := c.client.GetQueueUrl(ctx, &sqs.GetQueueUrlInput{
+		QueueName: aws.String(queueName),
+	})
+
+	if err != nil {
+		return nil, errors.Wrap(err, errors.New("error resolving sqs queue url"))
+	}
+
+	if result == nil || result.QueueUrl == nil {
+		return nil, errors.New("error resolving sqs queue url")
```
Comment (cristiandanner) [inline aws/aws-sdk-go.v2/client/sqs/client.go:69]: error message suggestion: sqs queue url not found

File: aws/aws-sdk-go.v2/client/sqs/client.go (hunk 1 above)
Comment (aleal) [reply aws/aws-sdk-go.v2/client/sqs/client.go:31]: it is a cache...

File: aws/aws-sdk-go.v2/client/sqs/client.go (hunk 2 above)
Comment (aleal) [reply aws/aws-sdk-go.v2/client/sqs/client.go]: changed for something more meaningful

**Review by cristiandanner — APPROVED** [review]

**Pull request merged**